### PR TITLE
enhance: highlight questions with errors in stack and block creation steps

### PR DESCRIPTION
**Pull request by sjschlapbach**
Screenshot 2024-08-13 at 17 11 09

### CODE REVIEW

File: apps/frontend-manage/src/components/sessions/creation/StackCreationErrors.tsx [hunk 1]
```diff
@@ -17,13 +17,43 @@ function StackCreationErrors({ errors }: StackCreationErrorsProps) {
...
+            )}: ${error}`}</li>
+          )
+      )}
+      {[...(errors.hasSampleSolutions ?? [])].flatMap(
+        (error: string, ix: number) =>
+          error && (
+            <li key={`error-questionId-hasSampleSolutions-${ix}`}>{`${t(
               'shared.generic.elementN',
               {
                 number: ix + 1,
```
[the remark this answers is not in the document]
Comment (sjschlapbach) [reply apps/frontend-manage/src/components/sessions/creation/StackCreationErrors.tsx:59]: we want to remove all empty elements / undefined elements within the existing array, which should not be mapped to list entries. Therefore the flatmap is used

File: apps/frontend-manage/src/components/sessions/creation/StackCreationErrors.tsx [hunk 2]
```diff
@@ -17,13 +17,43 @@
...
           error && (
-            <li key={`error-questionId-${ix}`}>{`${t(
+            <li key={`error-questionId-elementId-${ix}`}>{`${t(
+              'shared.generic.elementN',
+              {
+                number: ix + 1,
+              }
+            )}: ${error}`}</li>
+          )
+      )}
```
[the remark this answers is not in the document]
Comment (sjschlapbach) [reply apps/frontend-manage/src/components/sessions/creation/StackCreationErrors.tsx:30]: same as above

File: apps/frontend-manage/src/components/sessions/creation/StackCreationErrors.tsx [hunk 3]
```diff
@@ -17,13 +17,43 @@
...
+        (error: string, ix: number) =>
+          error && (
+            <li key={`error-questionId-types-${ix}`}>{`${t(
+              'shared.generic.elementN',
+              {
+                number: ix + 1,
+              }
+            )}: ${error}`}</li>
+          )
+      )}
```
[the remark this answers is not in the document]
Comment (sjschlapbach) [reply apps/frontend-manage/src/components/sessions/creation/StackCreationErrors.tsx:52]: same as above

**Comment by rschlaefli** [comment]
@coderabbitai generate unit testing code for this file.

**Pull request merged**